HW3

BoLjshOJ opened the pull request:
HW3 done

BoLjshOJ committed:
HW 3
init commit

BoLjshOJ committed:
HW 3 Перенести приложение на Spring Boot
-Перенес приложение проведения опросов
-Перенес все свойства в application.yml
-Сделал собственный баннер для приложения
-Перенес тесты и использовал spring-boot-test-starter

stvort commented on src/test/java/ru/otus/springFramework/dao/StudentDaoSimpleTest.java:
```diff
@@ -1,23 +1,25 @@
-        Student one = new Student("Petr", "Petrov", messageService);
-        assertThat(one.getName()).isEqualToIgnoringCase("petr");
-        assertThat(one.getSurname()).isEqualToIgnoringCase("petrov");
+        Student student = new Student("name", "surname", messageService);
```
Видимо в прошлый раз не заметил. Не нужно, чтобы сущности что-то знали о сервисах и уж тем более хранили на них ссылку. Если нужно локализованно переводить сущности в строку то лучше сделать для этого сервис. EntityPrinter или что-то подобное

BoLjshOJ replied:
Resolved by b849f6c75677ed4a2e47924354b861829b676c97

stvort commented on src/test/java/ru/otus/springFramework/dao/StudentDaoSimpleTest.java:
```diff
@@ -1,23 +1,25 @@
     @Test
-    void createStudent() {
+    public void createStudent() {
         messageService = mock(MessageService.class);
```
Т.к. тут у нас ```@SpringBootTest``` то мок messageService лучше сделать через ```@MockBean```. А то получается подняли контекст и вообще им не воспользовались

BoLjshOJ replied:
Resolved by b849f6c75677ed4a2e47924354b861829b676c97

BoLjshOJ committed:
HW3
1. Видимо в прошлый раз не заметил. Не нужно, чтобы сущности что-то знали о сервисах и уж тем более хранили на них ссылку. Если нужно локализованно переводить сущности в строку то лучше сделать для этого сервис. EntityPrinter или что-то подобное
- реализовал EntityPrinter
2. Т.к. тут у нас @SpringBootTest то мок messageService лучше сделать через @MockBean. А то получается подняли контекст и вообще им не воспользовались
-Из-за реализации EntityPrinter'a отпала необходимость мокать сервис

stvort commented on src/test/java/ru/otus/springFramework/dao/StudentDaoSimpleTest.java:
```diff
@@ -1,23 +1,21 @@
+    public void createStudent() {
+        Student student = new Student("name", "surname");
+        assertThat(student.getName()).isEqualToIgnoringCase("name");
+        assertThat(student.getSurname()).isEqualToIgnoringCase("surname");
```
Теперь это вообще не относится к StudentDaoSimple. Этот тест должен был бы выглядеть как-то так:
```
    @Autowired
    private StudentDao studentDao;
    @MockBean
    private IOService ioService;
    @Test
    public void createStudent(){
        Student studentExpected = new Student("Vasya", "Uber");
        given(ioService.getStringFromSonsole()).willReturn(studentExpected.getName()).willReturn(studentExpected.getSurname());
        Student studentActual = studentDao.createStudent();
        assertThat(studentActual.getName()).isEqualTo(studentExpected.getName());
        assertThat(studentActual.getSurname()).isEqualTo(studentExpected.getSurname());
    }
```

BoLjshOJ replied:
resolved by 9271a79fc0043b81037f8ba5b89d568f8d36469e

BoLjshOJ committed:
HW3
-update test for StudentDaoSimple
